app/src/main/java/com/example/ncov19traking/data/NCoVRepository.kt

```diff
@@ -1,86 +1,114 @@
 package com.example.ncov19traking.data
 
-import android.content.Context
 import android.util.Log
-import com.example.ncov19traking.api.NCoVApiAdapter
-import com.example.ncov19traking.models.NCoVInfo
-import com.example.ncov19traking.models.NCoVInfoYesterday
-import com.example.ncov19traking.models.NumbersByCountry
-import com.example.ncov19traking.models.Timeline
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.ncov19traking.api.*
+import com.example.ncov19traking.models.*
 import java.io.IOException
 
-class NCoVRepository(private val context: Context) {
+class NCoVRepository(nCoVDataBase: NCoVDataBase) {
 
-    private val nCoVDao = NCoVDataBase.getDataBase(context).nCoVDao()
-    private val countryDao = NCoVDataBase.getDataBase(context).countryDao()
-    private val globalHistoricalDao = NCoVDataBase.getDataBase(context).globalHistoricalDao()
+    private val nCoVDao = nCoVDataBase.nCoVDao()
+    private val countryDao = nCoVDataBase.countryDao()
+    private val globalHistoricalDao = nCoVDataBase.globalHistoricalDao()
+    private var errorResponse = MutableLiveData<ErrorBody>()
 
     suspend fun getAllCases(): NCoVInfo {
         return try {
-            refreshAllCases()
-            nCoVDao.load().first()
+            when (val response = ApiResponse.create(NCoVApiClient.nCoVApi.getGeneralNumbers())) {
+                is ApiSuccessResponse -> {
+                    nCoVDao.save(response.data)
+                    nCoVDao.load().first()
```

Comment:
Still repeated code `nCoVDao.load().first()` and `nCoVDao.load()` in the hole method